cargo-shuttle/src/lib.rs

```diff
@@ -588,6 +586,10 @@ impl Shuttle {
 
             if !response.success {
                 error!(error = response.message, "failed to load your service");
+                // TODO: we must kill the rest of the runtimes when we'll start multiple services from a workspace,
+                // but now this is not concerning given we're not supporting more than one services.
+                provisioner_server.abort();
+                runtime.kill().await?;
```

Comment:
For local runs we do actually start all the services we find currently. But perhaps we can just abort all the other handles in the joinset when one fails? :thinking:

Reply:
Addressed this here: https://github.com/shuttle-hq/shuttle/pull/805/commits/949cccb1d4e6756b29443293237640772d3754a5. Testing works well, all the previous successfully started runtimes are aborted and the processes reaped when a runtime's service panicks, including the current runtime.